crates/roc_std/src/roc_list.rs

```diff
@@ -103,6 +103,33 @@ impl<T> RocList<T> {
         self.len() == 0
     }
 
+    pub fn is_unique(&self) -> bool {
+        if let Some(storage) = self.storage() {
+            storage.is_unique()
+        } else {
+            // If there is no storage, this list is empty.
+            // An empty list is always unique.
+            true
+        }
+    }
+
+    pub fn is_readonly(&self) -> bool {
+        if let Some(storage) = self.storage() {
+            storage.is_readonly()
+        } else {
+            false
+        }
+    }
+
+    // Marks a list as readonly. This means that it will be leaked.
+    // For constants passed in from platform to application, this may be reasonable.
+    // Marked unsafe because it should not be used lightly.
+    pub unsafe fn set_readonly(&self) {
```

Comment:
it looks like this and the similar function in RocStr are not used outside of tests. Would it be better to do this work in some test helper rather than adding this to the Rusty API?

Reply:
I specially want this function, as mention in my use case above. That is why I added and exposed it.

Reply:
thanks for clarifying! 👍